libcnb-cargo/src/package/error.rs

```diff
@@ -129,6 +122,15 @@ pub(crate) enum Error {
 
     #[error("I/O error while calculating directory size\nPath: {0}\nError: {1}")]
     CalculateDirectorySize(PathBuf, std::io::Error),
+
+    #[error("Failed to spawn Cargo command\nError: {0}")]
+    SpawnCargoCommand(std::io::Error),
+
+    #[error("Unexpected Cargo exit status while attempting to read workspace root\nExit Status: {0}\nExamine Cargo output for details and potential compilation errors.")]
+    CargoCommandFailure(String),
```

Comment:
Not sure why this is using a `String` here. We should just pass the `ExitStatus` in here to pass around actual data instead of string representations meant for humans (as we've done in `FindCargoWorkspaceError`).

This should work and gives the same output as before, moving the display logic where it IMO belongs:

```suggestion
    #[error("Unexpected Cargo exit status while attempting to read workspace root\nExit Status: {}\nExamine Cargo output for details and potential compilation errors.", exit_status_or_unknown(.0))]
    CargoCommandFailure(ExitStatus),
```

(Note that this will not compile as long as `exit_status_or_unknown` takes ownership of the passed `ExitCode`, see comment there)